libkbfs/folder_branch_ops.go

```diff
@@ -4710,27 +4694,22 @@ func (fbo *folderBranchOps) rekeyLocked(ctx context.Context,
 			// The rekey bit was already set, and there's nothing else
 			// we can to do, so don't put any new revisions.
 			fbo.log.CDebugf(ctx, "No further rekey possible by this user.")
-			return nil
+			return RekeyResult{
+				DidRekey:      false,
+				NeedsPaperKey: false,
+			}, nil
 		}
 
 		// Rekey incomplete, fallthrough without early exit, to ensure
 		// we write the metadata with any potential changes
 		fbo.log.CDebugf(ctx,
 			"Rekeyed reader devices, but still need writer rekey")
 
-	case NeedOtherRekeyError:
-		stillNeedsRekey = true
-	case NeedSelfRekeyError:
+	case NeedOtherRekeyError, NeedSelfRekeyError:
 		stillNeedsRekey = true
 
 	default:
-		if err == context.DeadlineExceeded {
-			fbo.log.CDebugf(ctx, "Paper key prompt timed out")
-			// Reschedule the prompt in the timeout case.
-			stillNeedsRekey = true
-		} else {
-			return err
-		}
+		return RekeyResult{}, err
```

Comment:
Even so, the FSM uses a 0 delay in the error case, which is going to be annoying to someone who purposely ignored the popup.  (Before it retried after the 10 minute wait.)

I guess I can see the logic in retrying no matter what, I'm just worried that's going to make the logs harder to debug and put unnecessary strain on other parts of the system.  I still lean towards explicitly checking codes, but if you prefer this way, I'm ok with trying it.  But I do think specifically in the timeout case, we shouldn't retry immediately.